api/src/test/java/org/openmrs/module/fhir/api/PersonServiceTest.java

```diff
@@ -195,4 +192,43 @@ public void updateperson_shouldUpdateOmsPerson() throws Exception {
 		Assert.assertEquals(updatedPerson.getVoided(), false);
 		Assert.assertEquals(updatedPerson.getGender(), "F");
 	}
+
+	/**
+	 * @verifies(value="make person void", method="retirePerson(String))
+	 */
+	@Test
+	public void retirePerson_shouldMakePersonVoid() {
+		String personId = "dagh524f-27ce-4bb2-86d6-6d1d05312bd5";
+		org.openmrs.Person person = Context.getPersonService().getPersonByUuid(personId);
+		assertFalse(person.isVoided());
+		Context.getService(PersonService.class).retirePerson(personId.toString());
+		person = Context.getPersonService().getPersonByUuid(personId);
+		assertTrue(person.isVoided());
+	}
+
+	/**
+	 * @verifies(value="throw ResourceNotFoundException if person with given id not found", method="retirePerson(String))
+	 */
+	@Test(expected = ResourceNotFoundException.class)
+	public void retirePerson_shouldthrowResourceNotFoundExceptionIfPersonWithGivenIdNotFound() {
+		String personId = "--Not exists--";
+		Context.getService(PersonService.class).retirePerson(personId.toString());
+		fail("attempt to void non existent user should throw an exception");
+	}
+
+	/**
+	 * @verifies(value="do nothing if person already void", method="retirePerson(String))
+	 */
+	@Test
+	public void retirePerson_shouldDoNothingIfPersonAlreadyVoid() {
+		//No idea how to test it
```

Comment:
In FHIR there is no concept called retire. If you delete a resource through fhir, it will no longer available in the system. This is fine for now :+1: